exposed-money/src/main/kotlin/org/jetbrains/exposed/sql/money/CurrencyColumnType.kt

```diff
@@ -12,23 +14,62 @@ import javax.money.Monetary
  * @author Vladislav Kisel
  */
 @Suppress("MagicNumber")
-class CurrencyColumnType : VarCharColumnType(3) {
+class CurrencyColumnType : ColumnType<CurrencyUnit>() {
```

Comment:
Why do some column types change but not others?
For example, `EncryptedVarCharColumnType` still inherits from `VarCharColumnType`, rather than something like `ColumnType<String>`.

Reply:
Because both `EncryptedVarCharColumnType` and `VarCharColumnType` accept only `String` input type and they share some logic. `CurrencyColumnType` accepts only `CurrencyUnit` input type.

```
fun varchar(name: String, length: Int, collate: String? = null): Column<String> = registerColumn(
        name,
        VarCharColumnType(length, collate)
    )

fun Table.encryptedVarchar(name: String, cipherTextLength: Int, encryptor: Encryptor): Column<String> =
    registerColumn(name, EncryptedVarCharColumnType(encryptor, cipherTextLength))
```

`fun Table.currency(name: String): Column<CurrencyUnit> = registerColumn(name, CurrencyColumnType())`